src/main/java/nextmainfocus/account/AccountController.java

```diff
@@ -22,76 +19,39 @@
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
-import nextmainfocus.security.entity.Account;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
 
 @CrossOrigin(origins = "${client.url}")
 @RestController
 @RequestMapping("api/v1/")
-public class SecurityController {
+public class AccountController {
 
-	private static final Logger logger = LoggerFactory.getLogger(SecurityService.class);
+	private static final Logger logger = LoggerFactory.getLogger(AccountController.class);
 
 	@Autowired
-	SecurityService securityService;
+	AccountService securityService;
 
 	@Autowired
 	private InMemoryUserDetailsManager inMemoryUserDetailsManager;
 
 	@Autowired
 	public PasswordEncoder passwordEncoder;
 
-	@PostMapping(value = "users/", produces = "application/json")
-	public ResponseEntity<?> createUser(@RequestBody Account account) {
-		logger.info("Received request to create a new user: " + account.toString());
-
-		HttpStatus status;
-
-		try {
-			Account encodedUser = new Account();
-			encodedUser.setUsername(account.getUsername());
-			encodedUser.setPassword(passwordEncoder.encode(account.getPassword()));
-			encodedUser.setRole(account.getRole());
-
-			Account newDbAccount = securityService.createUser(encodedUser);
-
-			if (newDbAccount != null) {
-				UserDetails newAccount = getUserDetailsWithEncodedPswd(account);
-				inMemoryUserDetailsManager.createUser(newAccount);
-
-				JSONObject body = new JSONObject();
-				body.put("id", newDbAccount.getId());
-				body.put("username", newDbAccount.getUsername());
-				body.put("role", newDbAccount.getRole());
-				status = HttpStatus.OK;
-
-				logger.info(account.getUsername() + " was created");
-
-				return new ResponseEntity<>(body, status);
-			} else {
-				JSONObject body = new JSONObject();
-				body.put("code", "INTERNAL_SERVER_ERROR");
-				body.put("message", "Oops! User creation failed.");
-				status = HttpStatus.INTERNAL_SERVER_ERROR;
-
-				logger.error(account.getUsername() + " already presents in the database");
-				return new ResponseEntity<>(body.toString(), status);
-			}
-		} catch (Exception exception) {
-			JSONObject body = new JSONObject();
-			body.put("code", "INTERNAL_SERVER_ERROR");
-			body.put("message", "Oops! User creation failed.");
-			status = HttpStatus.INTERNAL_SERVER_ERROR;
+	private UserDetails getUserDetailsWithEncodedPassword(Account account) {
+		return org.springframework.security.core.userdetails.User.withUsername(account.getUsername())
+				.password(passwordEncoder.encode(account.getPassword())).roles(account.getRole()).build();
+	}
 
-			logger.error(account.getUsername() + " was not created due to:");
```

Comment:
do we still have logging of errors somewhere?

Reply:
This message is present. Methods are rearranged.